packages/dashboard/src/components/rmf-app/rmf-app.tsx

```diff
@@ -230,6 +232,16 @@ function TaskContextsProvider(props: React.PropsWithChildren<{}>): JSX.Element {
   return <TasksContext.Provider value={tasks}>{props.children}</TasksContext.Provider>;
 }
 
+function RmfHealthContextsProvider(props: React.PropsWithChildren<{}>): JSX.Element {
+  const healthManager = new RmfHealthStateManager();
```

Comment:
This creates a new manager each render, you need to `useState` or `useMemo` here.